addons/web/views/webclient_templates.xml

```diff
@@ -598,6 +598,21 @@
         </t>
     </template>
 
+    <template id="web.app_auth" name="Accept app">
+        <t t-call="web.login_layout">
+            <form class="oe_login_form" role="form" method="post" action="/web/oauth">
+                <input type="hidden" name="csrf_token" t-att-value="request.csrf_token()"/>
+                <input type="hidden" name="redirect" t-att-value="redirect"/>
+                <input type="hidden" name="scope" t-att-value="scope"/>
+
+                <t t-esc="scope" /> wants to access Odoo.
+
+                 <button type="submit" name="do" value="accept" class="btn btn-link btn-sm btn-block">Accept</button>
+                 <button type="submit" name="do" value="refuse" class="btn btn-link btn-sm btn-block">Refuse</button>
```

Comment:
*reject* would be more suitable. Or allow/deny, since this is about granting an access. Also maybe the value could just be a flag e.g. no `value` on the "Reject" button? Not sure it works, but it would probably be easier to test reliably.

Alternatively the `reject` button could be a link to the `redirect` with the relevant parameter tacked on?

Reply:
I labelled them allow/deny.
The value of "allow" is now just "1" (whatever value would do).
I removed the value for the "deny".

The code behind is now `if do ... else`